src/widget/wbeatspinbox.cpp

```diff
@@ -0,0 +1,277 @@
+    if (conversionWorked) {
+        return dValue;
+    }
+
+    QString sIntPart, sFracPart, sNumerator, sDenominator;
+
+    if (text.contains(" ")) {
+        QStringList numberParts = text.split(" ");
+        sIntPart = numberParts.at(0);
+        sFracPart = numberParts.at(1);
+    } else if (text.contains("/")) {
+        sFracPart = text;
+    }
+
+    QStringList splitFraction = sFracPart.split("/");
+    sNumerator = splitFraction.at(0);
+    sDenominator = splitFraction.at(1);
+
+    return sIntPart.toDouble() + sNumerator.toDouble() / sDenominator.toDouble();
+}
+
+QValidator::State WBeatSpinBox::validate(QString& input, int& pos) const {
+    Q_UNUSED(pos);
+    if (input.contains(s_regexpBlacklist)) {
+        return QValidator::Invalid;
+    }
+    if (input.isEmpty()) {
+        return QValidator::Intermediate;
+    }
+
+    if (input.count(" ") > 1 || input.count("/") > 1) {
+        return QValidator::Invalid;
+    }
+
+    bool conversionWorked = false;
+    input.toDouble(&conversionWorked);
+    if (conversionWorked) {
+        return QValidator::Acceptable;
+    }
+
+    QString sIntPart, sFracPart, sNumerator, sDenominator;
+
+    if (input.contains(" ")) {
+        // Whole number + fraction, for example "1 1/2"
+        QStringList numberParts = input.split(" ");
+        sIntPart = numberParts.at(0);
+        sFracPart = numberParts.at(1);
+        // Whole number + trailing space
+        if (sFracPart.isEmpty()) {
+            return QValidator::Intermediate;
+        }
+    } else if (input.contains("/")) {
+        // Fraction without whole number, for example "1/2"
+        sFracPart = input;
+    }
+
+    if (!sIntPart.isEmpty()) {
+        conversionWorked = false;
+        sIntPart.toInt(&conversionWorked);
+        if (!conversionWorked) {
+            return QValidator::Invalid;
+        }
+    }
+
+    if (!sFracPart.contains("/")) {
+        return QValidator::Intermediate;
+    }
+    QStringList fracParts = sFracPart.split("/");
+    sNumerator = fracParts.at(0);
+    if (sNumerator.isEmpty()) {
+        // when deleting the numerator, for example "/32"
+        return QValidator::Intermediate;
+    }
+    sDenominator = fracParts.at(1);
+
+    conversionWorked = false;
+    sNumerator.toInt(&conversionWorked);
+    if (!conversionWorked) {
+        return QValidator::Invalid;
+    }
+
+    if (sDenominator.isEmpty()) {
+        return QValidator::Intermediate;
+    }
+    conversionWorked = false;
+    sDenominator.toInt(&conversionWorked);
+    if (!conversionWorked) {
+        return QValidator::Invalid;
+    }
+
+    return QValidator::Acceptable;
+}
+
+bool WBeatSpinBox::event(QEvent* pEvent) {
+    if (pEvent->type() == QEvent::ToolTip) {
+        updateTooltip();
+    } else if (pEvent->type() == QEvent::FontChange) {
+        const QFont& fonti = font();
+        qDebug() << "WBeatSpinBox::event QEvent::FontChange" << fonti.pixelSize() * m_scaleFactor;
```

Comment:
Now that I added a font size to the stylesheet in #940, this is showing that the font gets scaled twice, but nothing actually changes on screen. Putting an analogous debug line in WEffectSelector::event also shows that the font gets scaled twice, but that does not actually happen. I tried adding `setValue(value());` after scaling the font, but that did not do anything.